packages/config-service/src/services/validationService.ts

```diff
@@ -0,0 +1,65 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { GlobalConfig } from './globalConfig';
+
+export class ValidationService {
+  static startUp(envs: NodeJS.Dict<string>): void {
+    // validate mandatory fields and their types
+    Object.entries(GlobalConfig.ENTRIES).forEach(([entryName, entryInfo]) => {
+      if (entryInfo.required) {
+        if (!envs.hasOwnProperty(entryName)) {
```

Comment:
QQ: Shouldn't we check if the entry has a default value here?

```suggestion
        if (!envs.hasOwnProperty(entryName) && !entryInfo.defaultValue) {
```

Reply:
The `if` statement above checks if the field is required so the default value check is unnecessary IMO.